src/environment.jl

```diff
@@ -0,0 +1,42 @@
+default_environment_algorithm() = "exact"
+
+function environment(
+  ψ::AbstractITensorNetwork,
+  vertices::Vector;
+  alg=default_environment_algorithm(),
+  kwargs...,
+)
+  return environment(Algorithm(alg), ψ, vertices; kwargs...)
+end
+
+function environment(
+  ::Algorithm"exact",
+  ψ::AbstractITensorNetwork,
+  vertices::Vector;
+  contraction_sequence_alg="optimal",
+  kwargs...,
+)
+  ψ_reduced = Vector{ITensor}(subgraph(ψ, vertices))
+  sequence = contraction_sequence(ψ_reduced; alg=contraction_sequence_alg)
+  return ITensor[contract(ψ_reduced; sequence, kwargs...)]
+end
+
+function environment(
+  ::Algorithm"bp",
+  ψ::AbstractITensorNetwork,
+  verts::Vector;
+  (cache!)=nothing,
+  partitions=default_partitioning(ψ),
```

Comment:
It's a bit long, but I think I would prefer `partitioned_vertices` instead of `partitions`, to be more explicit that it is expecting groups of vertices (rather than a partitioned tensor network, for example).